### ISSUE TITLE
Enforce Custom PR Checks with a Status Check

### ISSUE THREAD

**Issue by bdemann** [demergent-labs/azle#2295]
See if we can add a github workflow that will check the pr description for things that need to be done, then make that test required

**Comment by bdemann** [comment]
https://chatgpt.com/share/674e227b-3484-8004-95f5-520745ffa69a

**Comment by bdemann** [comment]
This task is all but done in the linked pr, we just need to decide which direction we want to go and go with it. I would recommend the label route for the reasons stated above.

**Comment by bdemann** [comment]
One possible draw back to tagging is that the tags do show up on the all pr page, which might cause lots of clutter, but it also might give us a quick idea of which prs have had the custom checks done or not

**Comment by bdemann** [comment]
Though, maybe a hybrid approach would be good. For example, it would be nice to just always have a "added breaking changes to the release issue" tag, but it might also be nice to have a check that says "all markdown checks have been checked" and that way if there are some extra things that have to happen before the pr is merged but we don't want to make tags for those one off things, we just make sure that all of the checks are checked first
